src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBinderFactory.cs

```diff
@@ -152,7 +152,14 @@ private class DefaultModelBinderProviderContext : ModelBinderProviderContext
             {
                 _factory = factory;
                 Metadata = factoryContext.Metadata;
-                BindingInfo = factoryContext.BindingInfo;
+                BindingInfo = new BindingInfo
+                {
+                    BinderModelName = factoryContext.BindingInfo?.BinderModelName ?? Metadata.BinderModelName,
+                    BinderType = factoryContext.BindingInfo?.BinderType ?? Metadata.BinderType,
+                    BindingSource = factoryContext.BindingInfo?.BindingSource ?? Metadata.BindingSource,
+                    PropertyFilterProvider =
+                        factoryContext.BindingInfo?.PropertyFilterProvider ?? Metadata.PropertyFilterProvider,
+                };
```

Comment:
Unit test for this? Should use a mock provider to assert that these properties are set for a top-level model